src/popupmenu.c

```diff
@@ -846,35 +847,27 @@ pum_redraw(void)
 
 				if (cells < remaining)
 				    over_cell =  remaining - cells;
-				new_str = alloc(size + over_cell + 1 + utf_char2len(fcs_trunc));
-				if (!new_str)
-				    return;
-				memcpy(new_str, st, size);
-				ptr = new_str + size;
-				if (over_cell > 0)
-				{
-				    vim_memset(ptr, ' ', over_cell);
-				    ptr += over_cell;
-				}
-
-				if (fcs_trunc != NUL)
-				    ptr += (*mb_char2bytes)(fcs_trunc, ptr);
-				else
-				    *ptr++ = '>';
-
-				*ptr = NUL;
-				vim_free(st);
-				st = new_str;
-				cells = mb_string2cells(st, -1);
-				size = (int)STRLEN(st);
-				width = cells;
+				cells = mb_string2cells(st, size);
+				width = cells + over_cell + 1;
 			    }
 
 			    if (attrs == NULL)
 				screen_puts_len(st, size, row, col, attr);
 			    else
 				pum_screen_puts_with_attrs(row, col, cells,
 							      st, size, attrs);
+			    if (truncated)
+			    {
+				if (over_cell > 0)
+				    screen_fill(row, row + 1, col + cells,
+					    col + cells + over_cell, ' ', ' ', attr);
+				if (fcs_trunc != NUL)
```

Comment:
```suggestion
				if (fcs_trunc != NUL && fcs_trunc != '>')
```

Reply:
a bit strange.. We don't need to check what fcs is set to here right?.

Reply:
I just noticed for symmetry with line 779 where this is also used.
```
if (fcs_trunc != NUL && fcs_trunc != '>')
```

Reply:
because it's in RL mode `>` need convert to `<`

Reply:
> because it's in RL mode > need convert to <

I'm skeptical about implementing something like this in a single feature.
For example, should it be delegated to the 'fillchars' option? Or left to the drawing part?
I think it's best to discuss any specifications that seem complicated or tedious before implementing them.